src/components/measurements/index.js

```diff
@@ -59,10 +63,27 @@ const treeStrainPropertySelector = (state) => {
     // Only store properties of terminal strain nodes
     if (!node.hasChildren) {
       treeStrainVisibility[node.name] = tree.visibility[index];
-      treeStrainColors[node.name] = {
-        attribute: getTipColorAttribute(node, colorScale),
-        color: tree.nodeColors[index]
-      };
+      // Only store colors for visible strains since only measurmeents
+      // for visible strains will be displayed
+      if (isVisible(tree.visibility[index])) {
+        /*
+        * If the color scale is continuous, we want to group by the legend value
+        * instead of the specific strain attribute in order to combine all values
+        * within the legend bounds into a single group.
+        */
+        let attribute = getTipColorAttribute(node, colorScale);
+        if (colorScale.continuous) {
+          colorScale.visibleLegendValues.forEach((legendValue) => {
```

Comment:
>  This does mean we have to loop through all of the visible legend values for each visible strain in the tree to determine the matching legend value, which is definitely not ideal.

Two potential speed-ups here are to (a) use a `for` loop and `break` when you have a legend match and (b) add an `else` statement and only set `attribute = getTipColorAttribute(node, colorScale);` within that. You'd have to time things to really work out what gain you get here, but at least (a) seems like it would be beneficial.

Reply:
You could also use the short-circuiting `Array.find()` method, e.g. something like

```
const legendValue = colorScale.visibleLegendValues
  .find(legendValue => determineLegendMatch(legendValue, node, colorScale));

if (legendValue) attribute = legendValue;
```

Reply:
Thanks both! Resolved in 29fed88